src/x86/avx2.rs

```diff
@@ -196,20 +197,126 @@ pub unsafe fn _mm256_blendv_epi8(a:i8x32,b:i8x32,mask:__m256i) -> i8x32 {
     pblendvb(a,b,mask)
 }
 
-// TODO _mm_broadcastb_epi8
-// TODO _mm256_broadcastb_epi8
-// TODO _mm_broadcastd_epi32
-// TODO _mm256_broadcastd_epi32
-// TODO _mm_broadcastq_epi64
-// TODO _mm256_broadcastq_epi64
-// TODO _mm_broadcastsd_pd
-// TODO _mm256_broadcastsd_pd
+/// Broadcast the low packed 8-bit integer from `a` to all elements of
+/// the 128-bit returned value.
+#[inline(always)]
+#[target_feature = "+avx2"]
+#[cfg_attr(test, assert_instr(vpbroadcastb))]
+pub unsafe fn _mm_broadcastb_epi8(a: i8x16) -> i8x16 {
+    simd_shuffle16(a, i8x16::splat(0i8), [0u32; 16])
+}
+
+/// Broadcast the low packed 8-bit integer from `a` to all elements of
+/// the 256-bit returned value.
+#[inline(always)]
+#[target_feature = "+avx2"]
+#[cfg_attr(test, assert_instr(vpbroadcastb))]
+pub unsafe fn _mm256_broadcastb_epi8(a: i8x16) -> i8x32 {
+    simd_shuffle32(i8x32::from(a), i8x32::splat(0i8), [0u32; 32])
+}
+
+/// Broadcast the low packed 32-bit integer from `a` to all elements of
+/// the 128-bit returned value.
+#[inline(always)]
+#[target_feature = "+avx2"]
+#[cfg_attr(test, assert_instr(vpbroadcastd))]
+pub unsafe fn _mm_broadcastd_epi32(a: i32x4) -> i32x4 {
+    simd_shuffle4(a, i32x4::splat(0i32), [0u32; 4])
+}
+
+/// Broadcast the low packed 32-bit integer from `a` to all elements of
+/// the 256-bit returned value.
+#[inline(always)]
+#[target_feature = "+avx2"]
+#[cfg_attr(test, assert_instr(vpbroadcastd))]
+pub unsafe fn _mm256_broadcastd_epi32(a: i32x4) -> i32x8 {
+    simd_shuffle8(i32x8::from(a), i32x8::splat(0i32), [0u32; 8])
+}
+
+/// Broadcast the low packed 64-bit integer from `a` to all elements of
+/// the 128-bit returned value.
+#[inline(always)]
+#[target_feature = "+avx2"]
+#[cfg_attr(test, assert_instr(vpbroadcastq))]
+pub unsafe fn _mm_broadcastq_epi64(a: i64x2) -> i64x2 {
+    simd_shuffle2(a, i64x2::splat(0i64), [0u32; 2])
+}
+
+/// Broadcast the low packed 64-bit integer from `a` to all elements of
+/// the 256-bit returned value.
+#[inline(always)]
+#[target_feature = "+avx2"]
+#[cfg_attr(test, assert_instr(vbroadcastq))]
+pub unsafe fn _mm256_broadcastq_epi64(a: i64x2) -> i64x4 {
+    simd_shuffle4(i64x4::from(a), i64x4::splat(0i64), [0u32; 4])
+}
+
+/// Broadcast the low double-precision (64-bit) floating-point element
+/// from `a` to all elements of the 128-bit returned value.
+#[inline(always)]
+#[target_feature = "+avx2"]
+#[cfg_attr(test, assert_instr(movddup))]
+pub unsafe fn _mm_broadcastsd_pd(a: f64x2) -> f64x2 {
+    simd_shuffle2(a, f64x2::splat(0f64), [0u32; 2])
+}
+
+/// Broadcast the low double-precision (64-bit) floating-point element
+/// from `a` to all elements of the 256-bit returned value.
+#[inline(always)]
+#[target_feature = "+avx2"]
+#[cfg_attr(test, assert_instr(vbroadcastsd))]
+pub unsafe fn _mm256_broadcastsd_pd(a: f64x2) -> f64x4 {
+    simd_shuffle4(f64x4::from(a), f64x4::splat(0f64), [0u32; 4])
+}
+
 // TODO _mm_broadcastsi128_si256
```

Comment:
Not sure what to do with this. A quick `rg _mm_broadcastsi128_si256` in the clang source didn't yield any results.

Reply:
After some discussion on gitter, following clang and not including `_mm_broadcastsi128_si256`